test/convert_reinterpret.jl

```diff
@@ -167,6 +167,57 @@ end
         b = @inferred(convert(Array{Gray{N0f8}}, a))
         @test eltype(b) == Gray{N0f8}
     end
+
+    # Gray images wrapped by an OffsetArray.
+    A = rand(8,8)
+    for img in ( Gray.(A),
+                 Gray.(N0f8.(A)),
+                 Gray.(N0f16.(A)) )
+    imgo = OffsetArray(img, 0, 0)
+    s = @inferred(convert(OffsetArray{Gray{Float32},2,Array{Gray{Float32}}},imgo))
+    @test eltype(s) == Gray{Float32}
+    end
+
+    for img in ( Gray.(A),
+                 Gray.(N0f8.(A)),
+                 Gray.(N0f16.(A)) )
+    imgo = OffsetArray(img, 0, 0)
```

Comment:
probably would be a good idea to use something like `OffsetArray(img, -2, -1)` and then later test that the following is true

```julia
@test indices(s) === indices(imgo)
```